src/packages/cli/src/__tests__/doctor.test.ts

```diff
@@ -1,74 +1,86 @@
 
-describe('doctor', () => {
-  it('doctor should succeed when schema and db match', async () => {
-    const tmpDir = tempy.directory()
-    await copy({
-      from: path.join(__dirname, 'fixtures/example-project/prisma'),
-      to: tmpDir,
-      recursive: true,
-    })
-    const cwd = process.cwd()
-    process.chdir(tmpDir)
+const ctx: {
+  tmpDir: string
+  logs: string[]
+  fs: FSJetpack
+  mocked: { 'console.error': any; cwd: string }
+} = {} as any
 
-    const doctor = Doctor.new()
-    const errLog = []
-    const oldConsoleError = console.error
-    console.error = (...args) => {
-      errLog.push(...args)
-    }
-    const result = await doctor.parse([])
-    console.error = oldConsoleError
+beforeEach(() => {
+  ctx.logs = []
+  ctx.tmpDir = tempy.directory()
+  ctx.mocked = {} as any
+  ctx.mocked['console.error'] = console.error
+  ctx.mocked.cwd = process.cwd()
+  ctx.fs = FSJet.cwd(ctx.tmpDir)
+  console.error = (...args) => {
+    ctx.logs.push(...args)
+  }
+  process.chdir(ctx.tmpDir)
+})
 
-    assert.deepEqual(errLog, [`👩‍⚕️🏥 Prisma Doctor checking the database...`])
-    assert.equal(result, 'Everything in sync 🔄')
+afterEach(() => {
+  console.error = ctx.mocked['console.error']
+  process.chdir(ctx.mocked.cwd)
+})
 
-    process.chdir(cwd)
+it('doctor should succeed when schema and db do match', async () => {
+  ctx.fs.copy(path.join(__dirname, 'fixtures/example-project/prisma'), '.', {
+    overwrite: true,
   })
+  await expect(Doctor.new().parse([])).resolves.toEqual('Everything in sync 🔄')
+  expect(ctx.logs).toEqual([`👩‍⚕️🏥 Prisma Doctor checking the database...`])
+})
 
-  it('should fail when schema and db dont match', async () => {
-    const tmpDir = tempy.directory()
-    await copy({
-      from: path.join(__dirname, 'fixtures/schema-db-out-of-sync'),
-      to: tmpDir,
-      recursive: true,
-    })
-    const cwd = process.cwd()
-    process.chdir(tmpDir)
+it('should fail when db is missing', async () => {
+  ctx.fs.copy(path.join(__dirname, 'fixtures/schema-db-out-of-sync'), '.', {
+    overwrite: true,
+  })
+  ctx.fs.remove('dev.db')
 
-    const doctor = Doctor.new()
-    const errLog = []
-    const oldConsoleError = console.error
-    console.error = (...args) => {
-      errLog.push(...args)
-    }
-    let err
-    try {
-      const result = await doctor.parse([])
-    } catch (e) {
-      err = e
-    }
-    console.error = oldConsoleError
+  await expect(
+    Doctor.new().parse([]),
+  ).rejects.toThrowErrorMatchingInlineSnapshot(
+    `"P1003: SQLite database file doesn't exist"`,
+  )
+})
 
-    assert.equal(
-      stripAnsi(err.message),
-      `
+it('should fail when db is empty', async () => {
+  ctx.fs.copy(path.join(__dirname, 'fixtures/schema-db-out-of-sync'), '.', {
+    overwrite: true,
+  })
+  ctx.fs.write(path.join(ctx.tmpDir, 'dev.db'), '')
+
+  await expect(Doctor.new().parse([])).rejects
+    .toThrowErrorMatchingInlineSnapshot(`
+          "[91mP4001[39m
```

Comment:
Easier to write the test with jest async here if we don't run any transformations like strip ansi.

I think capturing the colour output is even interesting frankly, as we do care about, no?

If we really don't want to capture the ansi however, I suggest we support ENV variables to do that and/or passing options to `parse()` method and/or `new()` constructor.

Reply:
For that I suggest providing a custom snapshot serializer. I just did that in Client and it works great so far:
https://github.com/prisma/prisma/blob/new-tests/src/packages/client/jest.config.js#L21
https://github.com/prisma/prisma/blob/new-tests/src/packages/client/scripts/jestSnapshotSerializer.js

Reply:
Awesome. I've copied it for now but we should share that somehow... In a monorepo what is the best way to share code between packages? ... I guess more packages 😬 ?

Reply:
Haha yeah normally a package is the right solution :D